frontend/lib/uast/Builder.cpp

```diff
@@ -413,6 +443,22 @@ void Builder::doAssignIDs(AstNode* ast, UniqueString symbolPath, int& i,
   if (search != notedLocations_.end()) {
     CHPL_ASSERT(!search->second.isEmpty());
     idToLocation_[ast->id()] = search->second;
+
+    // Also map additional locations to ID.
+    #define CHPL_LOCATION_MAP(ast__, location__) \
+      if (auto x = ast->to##ast__()) { \
+        auto& m1 = CHPL_AST_LOC_MAP(ast__, location__); \
+        auto it = m1.find(x); \
+        if (it != m1.end()) { \
+          auto& m2 = CHPL_ID_LOC_MAP(ast__, location__); \
+          auto& id = x->id(); \
+          CHPL_ASSERT(m2.find(id) == m2.end()); \
+          m2[id] = it->second; \
```

Comment:
You're performing a search twice here. If you preserve the iterator for `m2.find(id)`, you can `emplace_hint` to make the insertion not need to perform a search.

https://en.cppreference.com/w/cpp/container/map/emplace_hint

Alternatively, consider matching the code for `idToLocation` and not bothering with the assertion, or an `id` variable.